db/model/room.js

```diff
@@ -33,6 +34,14 @@ module.exports = function room(seq, dataTypes) {
       },
       comment: 'Create a named room ',
     },
+    settings: {
+      type: dataTypes.ARRAY(dataTypes.JSONB),
+      allowNull: true,
+      validate: {
+        contains: u.validateSettingsArray,
+      },
+      comment: 'Key/Value pairs for user specific settings',
```

Comment:
How many elements do you expect a typical *whatever* settings object to have? As an array, you will always have to iterate over the array to find one element, which would def. be slower for large objects. I would urge you to reconsider storing as array unless you are also going to *enforce* that it can only contain some small maximum number of elements.